protocols/request-response/src/lib.rs

```diff
@@ -437,8 +430,9 @@ where
     /// Checks whether an outbound request initiated by
     /// [`RequestResponse::send_request`] is still pending, i.e. waiting
     /// for a response.
-    pub fn is_pending_outbound(&self, req_id: &RequestId) -> bool {
-        self.pending_responses.contains_key(req_id)
+    pub fn is_pending_outbound(&self, request_id: &RequestId) -> bool {
+        self.connected.iter()
+            .any(|(_, cs)| cs.iter().any(|c| c.pending_inbound_responses.contains(request_id)))
```

Comment:
I don't think that is a practical option, as going through all connections is very inefficient. The method was only provided on the API because it could be implemented efficiently. Short of removing it, we could, for example, require a `PeerId` to be given together with the `RequestId`. Then we could offer symmetrically `is_pending_inbound()`. What do you think?